pkg/controller/compute/kubernetes/aws_handler.go

```diff
@@ -49,14 +48,14 @@ func (r *AWSClusterHandler) provision(class *corev1alpha1.ResourceClass, instanc
 
 	// set class and claim references
 	resourceInstance.ClassRef = class.ObjectReference()
-	resourceInstance.ClaimRef = instance.ObjectReference()
+	resourceInstance.ClaimRef = claim.ObjectReference()
 
 	// create and save EKSCluster
 	cluster := &awscomputev1alpha1.EKSCluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace:       class.Namespace,
-			Name:            fmt.Sprintf("eks-%s", instance.UID),
-			OwnerReferences: []metav1.OwnerReference{instance.OwnerReference()},
+			Name:            fmt.Sprintf("eks-%s", claim.GetObjectMeta().UID),
```

Comment:
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L24
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L129
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L29

This is tangential to this PR, but it would be nice if our `ResourceClaim` interface embedded the existing `ObjectMetaAccessor` interface. Currently we define a seductively similar but not quite identical `GetObjectMeta()` method that returns a concrete `*ObjectMeta` rather than an `Object` interface.

If we embedded the `ObjectMetaAccessor` interface in `ResourceClaim` any struct with an embedded `ObjectMeta` would continue to satisfy `ResourceClaim`, and we'd get the bonus of the handy `GetUID()`, `GetNamespace()` etc helpers provided by `Object`.

Reply:
I definitely agree with this one, it makes the interface and consumption of the metadata much more clean.  I wasn't aware of this interface, so thank you for pointing it out.